pkg/apis/core/validation/validation.go

```diff
@@ -5337,8 +5340,9 @@ func ValidateResourceRequirements(requirements *core.ResourceRequirements, fldPa
 		// Validate resource quantity.
 		allErrs = append(allErrs, ValidateResourceQuantityValue(string(resourceName), quantity, fldPath)...)
 
-		if helper.IsHugePageResourceName(resourceName) {
-			limContainsHugePages = true
+		limContainsHugePages, err = validateResourceQuantityHugePageValue(resourceName, quantity, opts)
```

Comment:
This overwrites an existing `limContainsHugePages` `true` value with `false` if the last resource validated is not a hugepages resource. This would be more straightforward like this:

```go
if helper.IsHugePageResourceName(resourceName) {
  limContainsHugePages = true
  if err := validateResourceQuantityHugePageValue(resourceName, quantity, opts); err != nil {
    allErrs = append(allErrs, field.Invalid(fldPath, quantity.String(), err.Error()))
  }
}
```

Please add a unit test that would have caught this: pod limit that has hugepages, then several non-hugepages/non-cpu/non-memory limit (several are required because iteration order is non-deterministic)

Reply:
done

Reply:
not seeing the change... did you push your latest changes?